### DISCUSSION TITLE
What is best way to style components and pages in NextJS?

### DISCUSSION THREAD

**Discussion by M1ck0** [vercel/next.js#16745] [Help]
Hello and thank you for visiting this thread

I was wondering which way is the best for styling components and pages in `NextJS`?

I've been doing `NextJS` for a few months now and still can't decide, so I came here for help from the community.

Somewhere I read it is `CSS-in-JS`, somewhere it is `CSS Modules` or `SASS/LESS`

I have a project coming up and it would be great if you could tell me from your own experience which way is fast and where developer experience is good, and if there is a golden middle between performance and developer experience.

Thank you and happy coding :+1:

**Comment by leerob** [comment]
There isn't a _best_ way – it all depends on what you're trying to build and the requirements on your system/users/customers.

Almost every response will be opinionated in some fashion. With that being said, here are my rankings/opinions.

1. First and foremost, stick to what you know. That will likely always be your best choice
2. Utilize built-in CSS modules support from Next.js
3. If you need complex components, and don't want to build them yourself, check out [Chakra UI](https://chakra-ui.com/)
4. If you hate CSS-in-JS, and want more "low level" utilities, use [Tailwind CSS](https://tailwindcss.com/)

**Reply by ninadsonawane** [reply]
What about theme-ui?

**Reply by leerob** [reply]
I've since wrote an entire post on this: https://leerob.io/blog/css-with-react

**Reply by nandorojo** [reply]
I've used theme-ui before, it's solid. However, I would recommend the other options over it.

Stitches, Chakra or Tailwind for design
HeadlessUI and/or Radix for primitives

I build for Next.js and React Native in one repo, so I use Dripsy.

But if I were building for only web, I would probably use Stitches and Radix for a very custom app. And if I wanted a more out-of-the-box app that looks solid and is fast to make, I would use Chakra UI.

Chakra and Tailwind both have paid versions with code snippets too, which can help speed things up. I don't use them, but you might find them useful.
